doc/reference/trd-appid.md

```diff
@@ -0,0 +1,651 @@
+application identifier. If the verifier policy does not assign a
+process binary an application identifier then the kernel MUST NOT load
+or run that process.
+
+Consider these five use cases.
+
+  1. A process binary that has no application credentials: it only
+  runs on kernels that are willing to load applications without
+  credentials (e.g., research systems). The verifier policy defines
+  that process binaries with no credentials have a local application 
+  identifier of a SHA256 hash of the application binary. 
+
+  1. The verifier policy defines that the global application
+  identifier of a process is the public key used to sign its process
+  binary. Application credentials for a process binary consist of a
+  TBF header containing this public key as well as a signature for the
+  process binary signed with the associated private key. Before
+  verifiying a signature in a TBF header, the verifier policy decides
+  whether to it accepts the associated public key. The verifier policy
+  assigns a global application identifier as the public key in the TBF
+  header.
+
+  1. Multiple separate process binaries that run concurrently need to
+  be signed with a single public key. Each process binary is
+  identified by a unique identifier I. Application credentials for
+  these process binaries consist of a TBF header containing an ECDSA
+  public key, the unique identifier I, and a signature over the
+  process binary and I signed with the private key corresponding to
+  the public key in the header. The kernel decides whether to accept a
+  particular public key for verification. The verifier policy assigns
+  a global application identifier as the concatenation of the public
+  key and I.
+
+  1. A Tock system wants to load the same process binary in 
+  two different processes at the same time. It cannot. Every process
+  binary has a single application identifier, and Tock will not run
+  two processes with the same application identifier.
+
+  1. A Tock system wants to load the same application binary
+  in two different processes at the same time. The system administrator
+  installs two process binaries on the device, which contain the
+  same application binary. The process binaries have no credentials.
+  The verifier policy assigns a local application identifier to each
+  process binary based on its order in application flash.
+
+An application identifier provides an identity for an application
+binary. It allows the Tock kernel to know about the provenance and
+origin of the binary and make access control or security decisions
+based on this information. For example, a kernel may allow only
+applications whose credentials use a particular trusted public key to
+access restricted functionality, but restrict other applications to
+use a subset of available system calls.
+
+Application identifiers are distinct from process
+identifiers. An application identifier is per-application (persists
```

Comment:
What is a process identifier?

Reply:
The process identifier field in a process structure in Tock.